client/my-sites/site-monitoring/components/site-monitoring-tab-panel/index.tsx

```diff
@@ -12,29 +15,36 @@ export const tabs = [
 ];
 
 interface SiteMonitoringTabPanelProps {
-	children: React.ComponentProps< typeof TabPanel >[ 'children' ];
 	selectedTab?: SiteMonitoringTab;
 	className?: string;
-	onSelected?: ( tabName: SiteMonitoringTab ) => void;
 }
 
 export const SiteMonitoringTabPanel = ( {
-	children: renderContents,
 	selectedTab = 'metrics',
 	className,
-	onSelected,
 }: SiteMonitoringTabPanelProps ) => {
+	const siteSlug = useSelector( getSelectedSiteSlug ) as string;
 	return (
-		<TabPanel
-			initialTabName={ selectedTab }
+		<SectionNav
+			selectedText={ selectedTab }
 			className={ classnames( 'site-monitoring-tab-panel', className ) }
-			tabs={ tabs }
-			onSelect={ ( tabName ) => {
-				updatePageQueryParam( tabName as SiteMonitoringTab );
-				onSelected?.( tabName as SiteMonitoringTab );
-			} }
 		>
-			{ ( tab ) => renderContents( tab ) }
-		</TabPanel>
+			<NavTabs>
+				{ tabs.map( ( { name, title } ) => {
+					return (
+						<NavItem
+							key={ name }
+							path={
+								name === 'metrics'
+									? `/site-monitoring/${ siteSlug }`
+									: `/site-monitoring/${ siteSlug }/${ name }`
+							}
+							selected={ selectedTab === name }
+							children={ title }
```

Comment:
Setting the `children` prop like this works fine, just wanted to point out that it's more idiomatic (and less surprising for readers IMO) to do:

```tsx
<NavItem
  key={ name }
  { /* ... more props */ }
>
  { title }
</NavItem>